homeassistant/components/melcloud/climate.py

```diff
@@ -150,6 +150,14 @@ def __init__(self, device: MelCloudDevice, ata_device: AtaDevice) -> None:
         self._attr_unique_id = f"{self.api.device.serial}-{self.api.device.mac}"
         self._attr_device_info = self.api.device_info
 
+    async def async_update(self) -> None:
+        """Update state from MELCloud."""
+        await super().async_update()
+
+        # We can only check for vane_horizontal once we fetch the device data from the cloud
+        if self._device.vane_horizontal:
+            self._attr_supported_features |= ClimateEntityFeature.SWING_HORIZONTAL_MODE
```

Comment:
So if the supported feature can change, I would recommend having a `supported_features` property. If it can't change I would recommend having it in `async_added_to_hass`

Reply:
ah, thanks, I didn't know about `async_added_to_hass`. I don't think the feature can change, so I've modified the commit to use that method instead of `async_update` (and tested that it works fine too)